nomenclature/region_mapping_models.py

```diff
@@ -188,3 +197,34 @@ def validate_mapping(cls, v, values):
         if invalid:
             raise RegionNotDefinedError(invalid, v.file)
         return v
+
+    def apply(self, df: IamDataFrame) -> IamDataFrame:
+        all_dfs: List[IamDataFrame] = []
+        for model in df.model:
+            model_df = df.filter(model=model)
+
+            # If no mapping is defined the data frame is returned unchanged
+            if model not in self.mappings:
+                all_dfs.append(model_df)
+            # Otherwise we first rename, then aggregate
+            else:
+                cm = self.mappings[model]
+
+                # Rename
+                if cm.native_regions is not None:
+                    all_dfs.append(
+                        model_df.rename(region=cm.renaming_dict).filter(
```

Comment:
I'm wondering... We assume that the native-region-renaming is a 1-to-1-mapping, so first filtering, then renaming should be faster...

Reply:
Good point, changed